python/neuroglancer/pipeline/task_queue.py

```diff
@@ -99,16 +96,15 @@ def __init__(self):
             super(LookupError, self).__init__('Queue Empty')
 
     def __init__(self, n_threads=40, project=PROJECT_NAME, queue_name=QUEUE_NAME):
+        super(TaskQueue, self).__init__(n_threads) # creates self._queue
+
         self._project = 's~' + project # s~ means North America, e~ means Europe
         self._queue_name = queue_name
+        self._api = self._initialize_interface()
 
-        self._api = googleapiclient.discovery.build('taskqueue', 'v1beta2', credentials=google_credentials)
-
-        self._queue = Queue.Queue(maxsize=0) # infinite size
-        self._threads = ()
-        self._terminate = threading.Event()
-
-        self._start_threads(n_threads)
+    # This is key to making sure threading works. Don't refactor this method away.
+    def _initialize_interface(self):
```

Comment:
That's an interesting idea, but it would prevent using the ThreadedQueue by itself though to process arbitrary functions. If we don't expect to ever do that, we should use @abstractmethod